File: Aula05/Java/Heranca-Multipla.java
// Interface Animal
abstract class Animal {
    protected String nome;

    public Animal(String nome) {
        this.nome = nome;
    }

    abstract void emitirSom();
}

class Mamifero extends Animal {
    public Mamifero(String nome) {
        super(nome);
    }

    public void amamentar() {
        System.out.println(nome + " esta amamentando.");
    }

    @Override
    void emitirSom() {

    }
}

interface Ave {
    void voar();
}

class Morcego extends Mamifero implements Ave {
    public Morcego(String nome) {
        super(nome);
    }

    @Override
    public void emitirSom() {
        System.out.println("Morcego fazendo ruidos noturnos.");
    }

    @Override
    public void voar() {
        System.out.println(nome + " esta voando.");
    }

    public static void main(String[] args) {
        Morcego morcego = new Morcego("morcego");
        morcego.emitirSom();   
        morcego.amamentar();  
        morcego.voar();     
    }
}

File: Aula05/Python/Heranca-Multipla.py
class Animal:
    def __init__(self, nome):
        self.nome = nome

    def emitir_som(self):
        pass

class Mamifero(Animal):
    def __init__(self, nome):
        super().__init__(nome)

    def amamentar(self):
        print(f"{self.nome} esta amamentando.")

class Ave(Animal):
    def __init__(self, nome):
        super().__init__(nome)

    def voar(self):
        print(f"{self.nome} esta voando.")

class Morcego(Mamifero, Ave):
    def __init__(self, nome):
        super().__init__(nome)

    def emitir_som(self):
        print("Morcego fazendo ruidos noturnos.")

# Exemplo de uso
morcego = Morcego("Morcego")
morcego.emitir_som()  
morcego.amamentar()  
morcego.voar()   
